spring-integration-stomp/src/main/java/org/springframework/integration/stomp/AbstractStompSessionManager.java

```diff
@@ -224,6 +273,11 @@ public void handleException(StompSession session, StompCommand command, StompHea
 
 		@Override
 		public void handleTransportError(StompSession session, Throwable exception) {
+			if (exception instanceof ConnectionLostException) {
+				this.session = null;
+				AbstractStompSessionManager.this.connected = false;
+				scheduleReconnect();
```

Comment:
Never mind. 
Missed your point here about the `applicationEventPublisher.publishEvent` above with the similar code.
Will introduce `StompConnectionFailedEvent`...
And yes: `StompSessionConnectedEvent` as well for the `onSuccess()` case above...